content/app_mesh/appmesh-implementation/appmeshconfigartion.md

```diff
@@ -0,0 +1,239 @@
++++
+title = "App Mesh Configuration"
+description = "App Mesh deployment"
+weight = 1
++++
+
+
+**AWS App Mesh** is a service mesh based on the Envoy proxy. It standardizes how microservices communicate, giving you end-to-end visibility and helping to ensure high-availability for your applications. 
+
+hroughout this workshop we will use AWS App Mesh, to gain consistent visibility and network traffic controls on our microservices running in **Amazon EC2 with Fargate**.
+
+A service mesh is composed of two parts, a _control plane_ and a _data plane_. AWS App Mesh is a managed control plane so you as a user don’t need to install or manage any servers to run it. The data plane for App Mesh is the open source Envoy proxy. **_It is your responsability to add the Envoy proxy as a side car to each microservice you want to expose and manage via App Mesh._**
+
+In the next few sections, you will perform 2 broad set of tasks:
+
+ - You will use the AWS CDK to create and configure the App Mesh resources needed to represent the microservices available in your environment like virtual services and virtual nodes.
+
+ - You will add a docker container running the Envoy proxy to each app where you have microservices running. AWS App Mesh supports EC2, ECS and EKS and here we will show you how to add Envoy in ECS, without modifying your microservice source code. 
+
+ - You will add an Ingress (Virtual Gateway) and Virtual Routers, so external clients to the mesh can interact with the (virtual) services that were already brought into the mesh. 
+
+**_A key feature of Envoy worth mentioning_** is that it exposes a set of APIs. These APIs are leveraged by AWS App Mesh to dynamically configure Envoy’s routing logic, freeing developers from the tedious task of manually updating config files.
+
+Now every time you launch an Envoy enabled microservice, the Envoy proxy will contact the AWS App Mesh Management API to subscribe to resource information for Listeners, Clusters, Routes, and Endpoints. The connection from the Envoy proxy to the AWS App Mesh management API endpoint is held open, which allows AWS App Mesh to stream updates to the Envoy proxy as users introduce changes to either of the resources listed above.
+
+Let’s start by creating the service mesh. A service mesh is a logical boundary for network traffic between the services that resides within it. To enable the app mesh, we will configure the ecsdemo-platform app as follow:
+
+Open the terminal and execute, so we can open the document that we need to configure:
+```bash
+c9 ~/environment/ecsdemo-platform/cdk/app.py
```

Comment:
We don't install the c9 package, it's totally good to use this and I like it, but we just need to add it to the prerequisites software section. I think it's `npm i -g c9`